### ISSUE TITLE
Using the --override flag for customizations field results in an error

### ISSUE THREAD

**Issue by rjhawar** [PrefectHQ/prefect#7125]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Prefect documentation for this issue.
- [X] I checked that this issue is related to Prefect and not one of its dependencies.

### Bug summary

When using the deployment build cli command with --override flag for customizations will result in an unexpected extra arguments error. 

Slack thread:
[https://prefect-community.slack.com/archives/CL09KU1K7/p1662470411245529](https://prefect-community.slack.com/archives/CL09KU1K7/p1662470411245529)

### Reproduction

```python3
Create simple deployment:

Using the deployment cli:
resource_object = [
        { "op": "add", "path": "/spec/template/spec/resources", "value": 
            {"limits": 
                {"memory": "1Gi", "cpu": "500m"}
            }
        }]
deployment build --override customizations=resource_object
```

### Error

```python3
Got unexpected extra arguments
```

### Versions

```Text
Version:             2.5.0
API version:         0.8.2
Python version:      3.9.14
Git commit:          eac37918
Built:               Thu, Oct 6, 2022 12:41 PM
OS/Arch:             darwin/x86_64
Profile:             cloud
Server type:         cloud
```

**Comment by EmilRex** [comment]
It does seem like there's a limit to the complexity of data you can pass in an override. At least [the implementation](https://github.com/PrefectHQ/prefect/blob/39ad6f8/src/prefect/agent.py#L201-L208) doesn't appear to support setting list items (only dict keys). In this case, I think it would be better to create the deployment as a [Python object](https://docs.prefect.io/concepts/deployments/#create-a-deployment-from-a-python-object).

## Example
1. Copy the below into a file named `hello.py`

```python3
from prefect import flow, get_run_logger

@flow
def hello(name: str = "Marvin"):
    logger = get_run_logger()
    logger.info(f"Hello {name}!")
```

2. Create a new KubernetesJob block in the UI, filling in only the `name` value as `override-test`
3. Run the following command,

```shell
prefect deployment build hello.py:hello -n override-test -ib kubernetes-job/override-test --override customizations=\[\]
```

4. Now check out `hello-deployment.yaml` and see that our override is there as a string.

**Comment by serinamarie** [comment]
Hi @rjhawar, thank you for this issue! If I'm understanding you correctly, I believe we have resolved your issue with this PR: https://github.com/PrefectHQ/prefect/pull/8349 which was [released in 2.8.0](https://github.com/PrefectHQ/prefect/blob/main/RELEASE-NOTES.md#release-280).

We also have improved our documentation on [how to pass in customizations as an override option](https://docs.prefect.io/concepts/deployments/#deployment-build-options).

It would be great if you have a chance to verify whether this helps you to successfully pass in customizations! 🙂

**Comment by rjhawar** [comment]
@serinamarie I actually got around this issue using another method, but the fix looks good to me. Please go ahead and close this request.
